Source/WebCore/loader/MixedContentChecker.cpp

```diff
@@ -53,53 +53,44 @@ bool MixedContentChecker::isMixedContent(SecurityOrigin& securityOrigin, const U
     return !SecurityOrigin::isSecure(url);
 }
 
-static void logWarning(const Frame& frame, bool allowed, ASCIILiteral action, const URL& target)
+static void logWarning(const Frame& frame, bool blocked, const URL& target)
 {
-    const char* errorString = allowed ? " was allowed to " : " was not allowed to ";
-    auto message = makeString((allowed ? "" : "[blocked] "), "The page at ", frame.document()->url().stringCenterEllipsizedToLength(), errorString, action, " insecure content from ", target.stringCenterEllipsizedToLength(), ".\n");
+    const char* errorString = !blocked ? "automatically upgraded and should" : "blocked and must";
+    auto message = makeString((!blocked ? "" : "[blocked] "), "The page at ", frame.document()->url().stringCenterEllipsizedToLength(), " requested insecure content from ", target.stringCenterEllipsizedToLength(), ". This content was ", errorString, " be served over HTTPS.\n");
     frame.document()->addConsoleMessage(MessageSource::Security, MessageLevel::Warning, message);
 }
 
-bool MixedContentChecker::canDisplayInsecureContent(Frame& frame, SecurityOrigin& securityOrigin, ContentType type, const URL& url, AlwaysDisplayInNonStrictMode alwaysDisplayInNonStrictMode)
+bool MixedContentChecker::shouldUpgradeInsecureContent(SecurityOrigin& securityOrigin, ContentType type, const URL& url, FetchOptions::Mode mode, Initiator initiator)
```

Comment:
Can we point to the spec here? It might be helpful to do it of other algorithms if possible

It seems this is sort of https://w3c.github.io/webappsec-mixed-content/#upgrade-algorithm but different, why is it different?

Reply:
Yes, I'll make this explicit. Also, I see this patch is missing a fix I made here, I'll include that in the next version.